src/components/Map.js

```diff
@@ -1,20 +1,32 @@
 import React, { useContext, useState } from 'react';
 
 import GoogleMapReact from 'google-map-react';
+import { Dialog } from '@reach/dialog';
+import '@reach/dialog/styles.css';
 
 import { SearchResultsContext } from '../context/SearchResults';
 
 import Pin from './Pin';
+import DialogBody from './DialogBody';
 
 const Map = () => {
   const [loaded, setLoaded] = useState(false);
+  const [locationDetails, setLocationDetails] = useState(null);
   const { data } = useContext(SearchResultsContext);
   const { centerCoords, zoom } = data;
 
   const handleApiLoaded = () => {
     setLoaded(true);
   };
 
+  const handleClick = (location) => {
+    setLocationDetails(location);
+  };
+
+  const handleClose = () => {
+    setLocationDetails(null);
+  };
```

Comment:
You could also handle the hiding and showing with a boolean, but this also seems fine. This would be where you could utilize the `isOpen` (sorry, it's `isOpen` and not `show`) and `onDismiss` props.